services/api/kalvi/api/views/auth.py

```diff
@@ -1,15 +1,87 @@
+from rest_framework_simplejwt.tokens import RefreshToken
+from rest_framework.permissions import IsAuthenticated
+from django.utils.encoding import smart_str, force_bytes, DjangoUnicodeDecodeError
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from db.models import User
 
+# Generate stateless token for User
+def get_tokens_for_user(user):
+  refresh = RefreshToken.for_user(user)
+  return {
+      'refresh': str(refresh),
+      'access': str(refresh.access_token),
+  }
 
+#viewset class for sign up
 class SignUpEndPoint(APIView):
-    def post(self, request):
-        return Response("SIGNUP ENDPOINT")
+    renderer_classes = [UserRenderer]
+    def post(self, request, format=None):
+        serializer = SignUpEndPointSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()       
+        token = get_tokens_for_user(user)
+        return Response({'token' : token, 'msg': 'user registered'}, status=status.HTTP_201_CREATED)
 
-    def get(self, request):
-        return Response("SIGNUP ENDPOINT")
+#viewset class for sign in 
+class SignInEndPoint(APIView):
+    renderer_classes = [UserRenderer]
+    def post(self, request, format=None):
+        serializer = SignInEndPointSerializer(data=request.data)  
+        serializer.is_valid(raise_exception=True)
+        email = serializer.data.get('email')
+        password = serializer.data.get('password')
+        user = authenticate(email=email, password=password)
+        if user:
+            token = get_tokens_for_user(user)
+            return Response({'token' : token,'msg' : 'Login Successful'}, status=status.HTTP_200_OK)
+        else:
+            return Response({'errors' : {'non_field_errors' : ["Email or password is not valid"]}}, status=status.HTTP_404_NOT_FOUND)
 
 
-class SignInEndPoint(APIView):
-    def post(self, request):
-        return Response("SIGNIN ENDPOINT")
+#viewset class for changing password, only authenticated users could change password. 
+class UserChangePasswordView(APIView):
+    renderer_classes = [UserRenderer]
+    permission_classes = [IsAuthenticated]
+    def post (self, request, format=None):
+        serializer = UserChangePasswordSerialiser(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = request.user
+        user.set_password(serializer.data.get('password'))
+        user.save()
+        return Response({'msg' : 'Password changed successfully'}, status=status.HTTP_200_OK)
+
+#viewset class for profile view of a user with get method using access tokens.
+class UserProfileView(APIView):
+    renderer_classes = [UserRenderer]
+    permission_classes = [IsAuthenticated]
+    def get(self, request, format=None):
+        serializer = UserProfileSerialiser(request.user)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+
+#viewset class for sending email to reset password in case of forget password.
```

Comment:
`viewSet for initiating password reset via email for forgotten password.`

Reply:
put this in right way.